<launch>
  <arg name="device_ip" default="192.168.1.200" />
  <arg name="msop_port" default="2368"/>
  <arg name="difop_port" default="2369"/>
  <arg name="use_gps_ts" default="false" />
  <!--c16表示机械式16线雷达；c32表示机械式32线雷达 -->
  <arg name="lidar_type" default="c16"/>

    <!--c16_2表示16线垂直角度分辨率为2度的雷达，c16_1表示16线垂直角度分辨率为1.33度的雷达 -->
  <arg name="c16_type" default="c16_2"/>

   <!--c32_2表示32线垂直角度分辨率为1度的雷达，c32_1表示32线垂直角度分辨率为0.33度的雷达 -->
  <arg name="c32_type" default="c32_2"/>

   <!--3表示32线fpga为2.7\2.8\3.0的版本的雷达，2表示32线fpga为2.6的版本的雷达-->
  <arg name = "c32_fpga_type" default="3"/>
  <!-- 1表示xyzirt格式，2表示xyzi格式 -->
  <arg name= "pcl_type" default="1"/>

  <arg  name="frame_id" default="laser"/>


  <node pkg="lslidar_driver" type="lslidar_driver_node" name="lslidar_driver_node" output="screen">
    <!--param name="pcap" value="$(find lslidar_c16_driver)/pcap/xxx.pcap" /-->
    <param name="device_ip" value="$(arg device_ip)" />
    <param name="msop_port" value="$(arg msop_port)" />
    <param name="difop_port" value="$(arg difop_port)"/>
    <param name="lidar_type" value="$(arg lidar_type)"/>
    <param name="c16_type" value="$(arg c16_type)"/>
    <param name="c32_type" value="$(arg c32_type)"/>
    <param name="c32_fpga_type" value="$(arg c32_fpga_type)"/>
    <param name="pcl_type" value="$(arg pcl_type)"/>
    <param name="add_multicast" value="false"/>
    <param name="group_ip" value="224.1.1.2"/>
    <param name="rpm" value="600"/>
    <param name="use_gps_ts" value="$(arg use_gps_ts)"/>
    <param name="min_range" value="0.15"/>
    <param name="max_range" value="150.0"/>
    <param name="frequency" value="10"/>
    <param name="frame_id" value="$(arg frame_id)"/>
    <param name="distance_unit" value="0.25"/>
    <param name="scan_start_angle" value="0"/>       <!-- 起始角度可以设置为负值 例如 -15750  -->
    <param name="scan_end_angle" value="36000"/>     
    <param name="scan_num" value="8"/>
    <param name="publish_scan" value="true"/>
    <param name="pointcloud_topic" value="point_cloud_raw"/>
    <param name="coordinate_opt" value="false"/>
    <param name="is_filter" value="false"/>
    <param name="filter_distance_min" value="0.0"/>
    <param name="filter_distance_max" value="0.0"/>
    <param name="filter_intensity_min" value="0.0"/>
    <param name="filter_intensity_max" value="0.0"/>
  </node>

  <!--node pkg="rviz" type="rviz" name="rviz" args="-d $(find lslidar_driver)/rviz_cfg/lslidar.rviz"/-->
 
 <!--node pkg="tf" type="static_transform_publisher" name="laser_link_to_world" args="0 0 0 0 0 0 world laser_link 100" /-->
  
</launch>
